cns/requestcontroller/kubecontroller/cnsinteractor.go

```diff
@@ -0,0 +1,17 @@
+package kubecontroller
+
+import (
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+)
+
+// CNSInteractor implements the CNSClient interface.
```

Comment:
IMO the CNS Client interface should live in the `cns` package or a `cns/client` package as `Client`, and then an implementation, say `HTTPClient`, should live next to it. I think client implementations should live close to the API in which they are a client for. Then, instead of having a `kubecontroller.CNSInteractor`, your controller can just have a `cns.Client` (or `client.Client`, depending on where you put it). Then, at instantiation of your controller, you can choose the implementation; for the `main` function, it would be the `HTTPClient`, and for tests, it could be a `mockClient` so you don't start mixing unit tests and integration tests.

Reply:
Done, made an APIClient interface in the cns/client package, and a Client implementation in the cns/client/httpapi subpackage